engine/plot_trajectory.py

```diff
@@ -0,0 +1,138 @@
+    max_coords = nodes[size[0] - 1, size[1] - 1].get_m_coords()
+    xlim = [min_coords[0] - delta, max_coords[0] + delta]
+    ylim = [min_coords[1] - delta, max_coords[1] + delta]
+    return xlim, ylim
+
+def init():
+    """
+    A function used to draw a clear frame, called once before the first frame
+
+    Note: Parameter to Matplotlib's FuncAnimation class.
+    This function is called before the first frame.
+    """
+    circle_patch.center = (0, 0)
+    circle_patch_base.center = mission.mission_state.base_station.position
+    ax.add_patch(circle_patch)
+    ax.add_patch(wedge_patch)
+    ax.add_patch(circle_patch_base)
+    ax.add_patch(wedge_patch_base)
+    return circle_patch, wedge_patch
+
+def animate(i, m):
+    """
+    Plot elements that animated in every frame.
+
+    Note: Parameter to Matplotlib's FuncAnimation class.
+    This function is called at each frame.
+
+    Arguments:
+        i: integer representing frame number (next value in FuncAnimation frames attribute)
+        m: a Mission object 
+    """
+    x_coord = m.mission_state.robot.robot_state.truthpose[i, 0]
+    y_coord = m.mission_state.robot.robot_state.truthpose[i, 1]
+    circle_patch.center = (x_coord, y_coord)
+    wedge_patch.update({"center": [x_coord, y_coord]})
+    wedge_patch.theta1 = np.degrees(m.mission_state.robot.robot_state.truthpose[i, 2]) - 10
+    wedge_patch.theta2 = np.degrees(m.mission_state.robot.robot_state.truthpose[i, 2]) + 10
+    return circle_patch, wedge_patch
+
+def plot_sim_traj(m):
+    """
+    Plot the trajectory the robot took in it's completed mission [m].
+    Currently, this uses simulated information to plot a simulated trajectory
+    for our robot. 
+
+    Arguments:
+        m: a Mission object 
+    """
+    # Global parameters for animation functions
+    global mission, ax, circle_patch, wedge_patch, circle_patch_base, wedge_patch_base, anim
+
+    plt.style.use('seaborn-whitegrid')
+    x_coords = m.mission_state.robot.robot_state.truthpose[:, 0]
+    y_coords = m.mission_state.robot.robot_state.truthpose[:, 1]
+    fig, ax = plt.subplots()
+    ax.plot(x_coords, y_coords, '-b')
+    ax.plot(x_coords[0], y_coords[0], 'gx')
+    margin = 5
+    
+    mission = m
+    circle_patch = plt.Circle((5, 5), 1, fc="green")
+
+    wedge_patch = patch.Wedge(
+        (5, 1), 3, 100, 80, animated=True, fill=False, width=2, ec="g", hatch="xx"
+    )
+
+    if m.mission_state.control_mode == ControlMode.ROOMBA:
+        range = m.mission_state.roomba_radius + margin
+        init_x = m.mission_state.base_station.position[0]
+        init_y = m.mission_state.base_station.position[1]
+        plt.xlim([init_x-range, init_x+range])
+        plt.ylim([init_y-range, init_y+range])
+        circle = plt.Circle((init_x, init_y), m.mission_state.roomba_radius)
+        ax.add_patch(circle)
+
+    elif m.mission_state.control_mode != ControlMode.MANUAL:
+        goals = waypoints_to_array(m.mission_state.all_waypoints)
```

Comment:
I don't think this is used anywhere?

Reply:
hmm yeah its greyed out on mine too i guess it wasnt used before as well? will remove